pkg/utils/wait/wait_test.go

```diff
@@ -0,0 +1,242 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package wait
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+)
+
```

Comment:
Follow this example

```golang
type conditionMocker struct {
	expectElapse      time.Duration
	expectUndoneCount int
	expectErrorCount  int

	undoneCount int
	errorCount  int
}

func (c *conditionMocker) condition(ctx context.Context) (bool, error) {
	if c.expectElapse > 0 {
		time.Sleep(c.expectElapse)
	}

	if c.undoneCount < c.expectUndoneCount {
		c.undoneCount++
		return false, nil
	}

	if c.errorCount < c.expectErrorCount {
		c.errorCount++
		return false, fmt.Errorf("error times %d", c.errorCount)
	}

	return true, nil
}

func TestPoll_WithTimeout(t *testing.T) {
	cm := conditionMocker{
		expectElapse: time.Second,
	}

	timeout := 50 * time.Millisecond

	start := time.Now()
	err := Poll(context.Background(), cm.condition, WithTimeout(timeout))
	elapsed := time.Since(start)

	if err == nil || !errors.Is(err, context.DeadlineExceeded) {
		t.Fatalf("Poll() error = %v, want %v", err, context.DeadlineExceeded)
	}

	if elapsed < timeout {
		t.Fatalf("Poll() timeout = %v, want >= %v", elapsed, timeout)
	}

	if elapsed >= cm.expectElapse {
		t.Fatalf("Poll() timeout = %v, want < %v", elapsed, cm.expectElapse)
	}
}

func TestPoll_WithInterval(t *testing.T) {
	cm := conditionMocker{
		expectElapse:      time.Second,
		expectUndoneCount: 3,
	}

	interval := time.Second

	start := time.Now()
	err := Poll(context.Background(), cm.condition, WithInterval(interval))
	elapsed := time.Since(start)

	if err != nil {
		t.Fatalf("Poll() error = %v, wantErr %v", err, false)
	}

	if cm.undoneCount != cm.expectUndoneCount {
		t.Fatalf("Poll() called condition %d times, want %d times", cm.undoneCount, cm.expectUndoneCount)
	}

	if elapsed < time.Duration(cm.expectUndoneCount)*cm.expectElapse {
		t.Fatalf("Poll() elapsed = %v, want >= %v", elapsed, time.Duration(cm.expectUndoneCount)*cm.expectElapse)
	}
}

```

Reply:
The other tests are modified in this way

Reply:
I'm sorry, i didnt get you, can you explain me again? Should i modify all the test cases as per your example or only for TestPoll_WithInterval and TestPoll_WithTimeout. As of now, i modified the tests which can follow those example like TestPoll_Default, TestPoll_WithTimeout, TestPoll_WithInterval, TestPoll_EdgeCases, TestPoll_WithContinueOnError